src/Illuminate/Routing/RouteCollection.php

```diff
@@ -263,7 +196,9 @@ public function hasNamedRoute($name)
      */
     public function getByName($name)
     {
-        return isset($this->nameList[$name]) ? $this->nameList[$name] : null;
+        return $this->collection->first(function (Route $route) use ($name) {
+            return Arr::get($route->getAction(), 'as') === $name;
+        });
```

Comment:
Isn't this much slower than looking it up in a map? People may have many `route(...)` calls in their views.